cabal-install/Distribution/Client/BuildTarget.hs

```diff
@@ -498,15 +501,33 @@ resolveBuildTarget ppinfo opinfo userTarget =
         expected = [ (thing, got) 
                    | (_, MatchErrorExpected thing got)
                            <- map (innerErr Nothing) errs ]
-        nosuch   = [ (inside, thing, got, alts)
-                   | (inside, MatchErrorNoSuch thing got alts)
-                           <- map (innerErr Nothing) errs ]
+        nosuch   = Map.foldrWithKey genResults [] $ Map.fromListWith Set.union $
+          [ ((inside, thing, got), Set.fromList alts)
+          | (inside, MatchErrorNoSuch thing got alts) <- map (innerErr Nothing) errs
+          ]
+
+        genResults (inside, thing, got) alts acc = (
```

Comment:
Would be good to say what this chunk of code is trying to do. Something like, add a list of similarly named alternatives, giving up to three suggestions.